treeFormatting.js

```diff
@@ -33,30 +34,35 @@ function parenthesizeTree(tree, options){
 	}
 
 	function processNode(node){
-		//console.log(node);
 		var nonTerminal = (node.children instanceof Array) && node.children.length;
 		if (showNewCats && !parens.hasOwnProperty(node.cat)){
 			parens[node.cat] = ["["+node.cat+" ", "]"];
 		}
-		for (item in node){
-			if (item == "id" || item == "cat" || item == "children"){
-				continue;
-			}
-			if (node[item]){
-				if (toneTree.length == 0){
-					toneTree.push(item);
-				}
-				else{
-					toneTree[0] += '.' + item;
-				}
-			}
-			// console.log(item);
-			// console.log(node[item]);
-		}
+
 		var visible = invisCats.indexOf(node.cat) === -1 && parens.hasOwnProperty(node.cat);
 		if (nonTerminal) {
 			if (visible) {
-				parTree.push(parens[node.cat][0]);//pushes the right parens
+				if (node["func"] && node["silentHead"]){
+					parTree.push(parens[node.cat][0] + ".f.sh");
+				}
+				else if (node["func"]){
+					parTree.push(parens[node.cat][0] + ".f");
+				}
+				else if (node["silentHead"]){
+					parTree.push(parens[node.cat][0] + ".sh");
+				}
+				else{
+					parTree.push(parens[node.cat][0]);
+				}//pushes the right parens}
+
+				if(node.head && options.showHeads){
```

Comment:
This is pushing a * right after the opening parenthesis. We actually want it after the closing parenthesis.
Ex:
{(* a b c*)} should be {(a b c*)*}
(* (a b*) (* c*)) should be ( (a b*) (c*)* )